packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -33,6 +34,40 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
     return Math.min(licenseRetentionDays, DEFAULT_RETENTION_DAYS);
   };
 
+  // TODO: Refactor so i18n can interact with history without history itself being concerned about i18n
+  const getLocaleDictionary = async () => {
+    if (!strapi.plugin('i18n')) {
+      return {};
+    }
+
+    const locales = (await strapi.plugin('i18n').service('locales').find()) || [];
+    return locales.reduce(
+      (
+        acc: Record<string, NonNullable<HistoryVersions.HistoryVersionDataResponse['locale']>>,
+        locale: NonNullable<HistoryVersions.HistoryVersionDataResponse['locale']>
+      ) => {
+        acc[locale.code] = { name: locale.name, code: locale.code };
+
+        return acc;
+      },
+      {}
+    );
+  };
+
+  const getVersionStatus = async (
+    contentTypeUid: HistoryVersions.CreateHistoryVersion['contentType'],
+    data: HistoryVersions.CreateHistoryVersion['data']
+  ) => {
+    try {
+      const documentMetadataService = strapi.plugin('content-manager').service('document-metadata');
+      const meta = await documentMetadataService.getMetadata(contentTypeUid, data);
+
+      return documentMetadataService.getStatus(data, meta.availableStatus);
+    } catch (error) {
+      console.error(error);
+    }
```

Comment:
I'm curious to hear your thinking here to catch/silence the error

Reply:
I left a comment [here](https://github.com/strapi/strapi/pull/19737#discussion_r1521474444). I should test this again by throwing an error in the getMetadata function. I had a couple instances during development where I had an error tryin to get the status and it meant in the CM I couldn't create an entry. I don't think history should get in the way of creating content. Wdyt?

Reply:
Actually nvm seems to create the entry no matter what. I think I can remove this

Reply:
probably because it's called after awaiting next()